### PR TITLE
feat(evaluation): Add CJK tokenizer support for ROUGE-1 evaluation

### LINKED ISSUE #4122: `response_match_score` (ROUGE-1) is not effectively in Japanese (with manual tokenization)
**Describe the bug**
`response_match_score` uses `rouge-score`'s English-oriented tokenizer.
Japanese text without whitespace is treated as a single token, so scores are incorrect.
Because we have no built-in Japanese tokenization, we had to manually pre-tokenize the Japanese input and pass it as space-separated text.
This is a workaround but there is a bug..

**To Reproduce**
Minimal example (using pre-tokenized Japanese as a workaround):

```python
from google.adk.evaluation.final_response_match_v1 import _calculate_rouge_1_scores

candidate = "これ は テスト 候補 の 応答"
reference = "これ は テスト の 正解"
score = _calculate_rouge_1_scores(candidate, reference)
print(score)
```

```
Score(precision=0.0, recall=0.0, fmeasure=0.0)
```

Even in this case, the score depends on manual tokenization and does not reflect native Japanese text behavior.

**Expected behavior**
Japanese should be supported without requiring users to manually insert whitespace.
Ideally response_match_score should accept a tokenizer option or provide language-aware tokenization for ROUGE.

```
Score(precision=0.6666666666666666, recall=0.8, fmeasure=0.7272727272727272)
```

**Desktop (please complete the following information):**

- OS: macOS
- Python version(python -V): Python 3.13.8
- ADK version(pip show google-adk): v1.22.0

**Model Information:**

- Are you using LiteLLM: No
- Which model is being used(e.g. gemini-2.5-pro): N/A (offline metric)

**Additional context**

- The limitation comes from [rouge-score](https://pypi.org/project/rouge-score/) default tokenization.
    - https://github.com/google-research/google-research/blob/f6a8c2c254838f75b8429c5de456f2c1f2d8dc7c/rouge/tokenize.py#L51-L52
- We had to pass pre-tokenized Japanese to get any meaningful ROUGE-1 result, whichis not ideal.

### PR DESCRIPTION
**Pull request by maru0804**
## Summary

Fixes #4122

The default ROUGE tokenizer only recognizes ASCII alphanumeric characters (`[a-z0-9]`), causing ROUGE-1 scores to be **0.0** for CJK (Chinese, Japanese, Korean) text. This PR adds CJK language support through an opt-in tokenizer.

### New Features
- **`CJKTokenizer`**: A character-based tokenizer for CJK languages that:
  - Tokenizes CJK characters individually (1 char = 1 token)
  - Preserves word-based tokenization for ASCII alphanumeric
  - Removes CJK punctuation (U+3000-U+303F)
  - Skips other scripts (Greek, Cyrillic, fullwidth alphanumeric, etc.)

- **`RougeScoreCriterion`**: New criterion class to specify tokenizer options

### Modified
- **`RougeEvaluator`**: Updated to support custom tokenizers and log a warning (once) when CJK text is detected without proper tokenizer configuration

- **`ResponseEvaluator`**: Updated to pass `eval_metric` (including criterion) to `RougeEvaluator`

## Usage

```python
from google.adk.evaluation.eval_metrics import EvalMetric, RougeScoreCriterion

criterion = RougeScoreCriterion(threshold=0.8, tokenizer='cjk')
eval_metric = EvalMetric(
    metric_name='response_match_score',
    threshold=0.8,
    criterion=criterion,
)
```

## Backward Compatibility
- ✅ Default behavior unchanged (ASCII-only tokenization)
- ✅ Existing tests pass (365 evaluation tests)
- ✅ Warning logged when CJK detected without tokenizer

## Limitations (documented in docstrings)
- Fullwidth alphanumeric (Ａ-Ｚ, ０-９) are skipped
- Greek, Cyrillic, and other non-CJK scripts are skipped
- Character-based tokenization, not morphological analysis (for Japanese morphological analysis, consider MeCab)

## Test Coverage
Added 29 new tests covering:
- CJKTokenizer tokenization behavior
- ROUGE score calculation with/without CJK tokenizer
- Warning behavior (logged once per instance)
- Edge cases (empty strings, None, mixed text, punctuation)

### CODE REVIEW

File: src/google/adk/evaluation/final_response_match_v1.py [hunk 1]
```diff
@@ -14,29 +14,166 @@
...
+        while i < n and _is_ascii_alnum(text[i]):
+          i += 1
+        tokens.append(text[word_start:i])
+      else:
+        i += 1
+
+    return tokens
+
 
 class RougeEvaluator(Evaluator):
```
[the remark this answers is not in the document]
Comment (maru0804) [reply src/google/adk/evaluation/final_response_match_v1.py:116]: fix
https://github.com/google/adk-python/pull/4143/commits/a5bae8d2a68d22f2848a5272f7455b6d90ae7e7b

File: src/google/adk/evaluation/final_response_match_v1.py [hunk 2]
```diff
@@ -14,29 +14,166 @@
...
+        i += 1
+      elif _is_ascii_alnum(char):
+        word_start = i
+        while i < n and _is_ascii_alnum(text[i]):
+          i += 1
+        tokens.append(text[word_start:i])
+      else:
+        i += 1
+
+    return tokens
```
[the remark this answers is not in the document]
Comment (maru0804) [reply src/google/adk/evaluation/final_response_match_v1.py]: fix
https://github.com/google/adk-python/pull/4143/commits/a5bae8d2a68d22f2848a5272f7455b6d90ae7e7b

**Comment by rohityan** [comment]
Hi @maru0804, Thank you for your contribution! We appreciate you taking the time to submit this pull request. Your PR has been received by the team and is currently under review. We will provide feedback as soon as we have an update to share.

**Comment by rohityan** [comment]
Hi @ankursharmas , can you please review this.

**Comment by ankursharmas** [comment]
https://github.com/google/adk-python/pull/6292 already addresses the issue. We should close this PR.
